lib/ae_mdw/collection.ex

```diff
@@ -118,6 +119,38 @@ defmodule AeMdw.Collection do
     |> remove_dups()
   end
 
+  @spec generate_key_boundary(tuple()) :: key_boundary()
+  def generate_key_boundary(key) do
+    size = tuple_size(key)
+
+    (size - 1)..0
+    |> Enum.reduce({[], []}, fn i, {first, last} ->
+      {[get_min_key(elem(key, i)) | first], [get_max_key(elem(key, i)) | last]}
+    end)
+    |> then(fn {first, last} -> {List.to_tuple(first), List.to_tuple(last)} end)
```

Comment:
maybe we can try something like this to avoid indexes
```elixir
key
|> Tuple.to_list()
|> Enum.map(&{get_min_key(&1), get_max_key(&1)})
|> Enum.unzip()
|> then(fn {first, last} -> {List.to_tuple(first), List.to_tuple(last)} end)
```